### PR TITLE
feat: support no_std for alloy-genesis/alloy-serde

### PR DESCRIPTION
**Pull request by yjhmelody**
## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [x] Breaking changes

### CODE REVIEW

File: crates/serde/src/json_u256.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //! Json U256 serde helpers.
 
+#[allow(unused_imports)]
```
Comment (yjhmelody) [inline crates/serde/src/json_u256.rs:3]: Seems the clippy warnings are not correct, for no_std must import `String`/`ToString`

File: crates/serde/src/u64_hex.rs
Comment (mattsse) [inline crates/serde/src/u64_hex.rs:1]: why was this removed?

File: crates/genesis/src/lib.rs [hunk 2]
```diff
@@ -226,7 +230,7 @@ pub struct GenesisAccount {
         skip_serializing_if = "Option::is_none",
         deserialize_with = "deserialize_storage_map"
     )]
-    pub storage: Option<HashMap<B256, B256>>,
+    pub storage: Option<BTreeMap<B256, B256>>,
```
Comment (mattsse) [inline crates/genesis/src/lib.rs:233]: I actually prefer the BtreeMap change here so serialize/deserialize is consistent

File: crates/serde/src/json_u256.rs (hunk 1 above)
Comment (mattsse) [reply crates/serde/src/json_u256.rs:3]: you can feature gate these imports instead:

see also 

https://github.com/alloy-rs/alloy/blob/main/crates/eips/src/eip2930.rs#L7-L8

File: crates/serde/src/json_u256.rs [hunk 3]
```diff
@@ -175,6 +179,8 @@ where
 #[cfg(test)]
 mod test {
     use super::JsonU256;
+    #[allow(unused_imports)]
+    use alloc::{vec, vec::Vec};
```
Comment (mattsse) [inline crates/serde/src/json_u256.rs:183]: https://github.com/alloy-rs/alloy/blob/main/crates/eips/src/eip2930.rs#L7-L8

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive

smol nits, replace unused_imports with `#[cfg(not(feature = "std"))]`

File: crates/serde/src/json_u256.rs (hunk 3 above)
Comment (yjhmelody) [reply crates/serde/src/json_u256.rs]: I do this but it do not work.

File: crates/serde/src/u64_hex.rs
Comment (yjhmelody) [reply crates/serde/src/u64_hex.rs:1]: It's not used.

File: Cargo.toml [hunk 4]
```diff
@@ -94,8 +94,8 @@ thiserror-no-std = "2.0.2"
 url = "2.5"
 
 ## serde
-serde = { version = "1.0", features = ["derive"], default-features = false }
-serde_json = "1.0"
+serde = { version = "1.0", default-features = false, features = ["derive"] }
+serde_json = { version = "1.0" }
```
Comment (mattsse) [inline Cargo.toml:98]:
```suggestion
serde_json = "1.0"
```

**Review by mattsse — APPROVED** [review]
lgtm,

pending @DaniPopes

File: README.md [hunk 5]
```diff
@@ -101,7 +101,11 @@ contribution follows the linting rules and passes clippy.
 ## Note on `no_std`
 
 Because these crates are primarily json-rpc focused, we do not intend to support
-`no_std` for them at this time.
+`no_std` for most of them at this time. 
+But the following basic crates support `no_std`:
```
Comment (onbjerg) [inline README.md:105]:
```suggestion
The following crates support `no_std`:
```

nit

**Review by onbjerg — APPROVED** [review]
